internal/p2p/peermanager.go

```diff
@@ -368,20 +401,47 @@ func (m *PeerManager) prunePeers() error {
 	ranked := m.store.Ranked()
 	for i := len(ranked) - 1; i >= 0; i-- {
 		peerID := ranked[i].ID
+
 		switch {
 		case m.store.Size() <= int(m.options.MaxPeers):
 			return nil
 		case m.dialing[peerID]:
-		case m.connected[peerID]:
+		case m.isConnected(peerID):
```

Comment:
Ah, it used to be a map of id's to bools, which meant that you could just look at the value in the map. Now because we're tracking incoming and outgoing connections in this map differently, the function makes it more clear, I think.